### PR TITLE
test: add unit tests for shared utility functions (closes #7)

### LINKED ISSUE #7: [Testing] Add unit tests for @meridian/shared utility functions
## Problem

`packages/shared/src/utils.ts` contains `formatUsdAmount`, `parseUsdAmount`, `fromStroops`, and `shortenAddress` — all used across the frontend and API with no test coverage.

## Location in codebase

- `packages/shared/src/utils.ts`
- `packages/shared/src/utils.test.ts` (create)

## Tasks

- [ ] Add `vitest` to `packages/shared` devDependencies
- [ ] Add a `test` script: `vitest run --passWithNoTests`
- [ ] Write tests for `formatUsdAmount` — zero, typical, large values, decimal edge cases
- [ ] Write tests for `parseUsdAmount` — round-trip with `formatUsdAmount`
- [ ] Write tests for `fromStroops` — known conversion values
- [ ] Write tests for `shortenAddress` — standard G-address, short input

## Acceptance Criteria

- `pnpm test` passes with at least 12 assertions covering all four utilities
- Edge cases: zero, max BigInt, address shorter than 8 chars

### PR DESCRIPTION
**Pull request by Diyaaa-12**
## Summary

- Added unit tests for all four utility functions in `packages/shared/src/utils.ts` as specified in #7
- `fromStroops` — whole, fractional, zero, negative, max BigInt (asserts actual output value)
- `formatUsdAmount` — zero, typical, large, NaN, Infinity
- `parseUsdAmount` — round-trip, empty string, non-numeric, malformed
- `shortenAddress` — standard G-address, custom chars, short input (returns address unchanged when too short)
- Also added `fromStroops`, `formatUsdAmount`, and `parseUsdAmount` implementations which were not present in main
- Fixed `shortenAddress` to guard against short inputs

## Test plan

- [x] `pnpm --filter @meridian/shared test` — 45 tests passing
- [x] `pnpm --filter @meridian/shared typecheck` exits 0
- [x] No `any` types used

Closes #7

### CODE REVIEW

**Comment by Diyaaa-12** [comment]
Hi @collinsezedike, cleaned up and reopened as #300 with a single commit and no conflicts. All CI checks passing. Ready for review!

File: packages/shared/src/utils.test.ts [hunk 1]
```diff
@@ -158,13 +159,24 @@ describe("fromStroops", () => {
   it("handles negative exact value", () => {
     expect(fromStroops(-10_000_000n)).toBe("-1");
   });
+
+  it("handles max BigInt value without throwing", () => {
+    expect(() => fromStroops(9_007_199_254_740_991n)).not.toThrow();
```
Comment (collinsezedike) [inline packages/shared/src/utils.test.ts]: BigInt arithmetic never throws on large values, so this is always true regardless of what the function returns. A regression that corrupts the output while keeping the function alive passes silently. Assert the actual value instead: `expect(fromStroops(9_007_199_254_740_991n)).toBe("900719925.4740991")`.

File: packages/shared/src/utils.test.ts [hunk 2]
```diff
@@ -191,3 +207,17 @@ describe("parseUsdAmount", () => {
...
+    expect(shortenAddress("GABCDEFGHIJKLMNOPQRSTUVWXYZ234567ABCDEFGHIJKLMNOPQRSTUVWXYZ")).toBe("GABC...WXYZ");
+  });
+
+  it("respects a custom chars argument", () => {
+    expect(shortenAddress("GABCDEFGHIJKLMNOPQRSTUVWXYZ234567ABCDEFGHIJKLMNOPQRSTUVWXYZ", 6)).toBe("GABCDE...UVWXYZ");
+  });
+
+  it("handles a short input without crashing", () => {
+    expect(shortenAddress("GABC")).toBe("GABC...GABC");
+  });
```
Comment (collinsezedike) [inline packages/shared/src/utils.test.ts:219]: `"GABC".slice(-4)` wraps to index 0 and returns the full string, so the output is `"GABC...GABC"` - the `...` implies truncation when nothing was cut. The implementation should guard against this: `if (address.length <= chars * 2) return address;`. The test should then assert `toBe("GABC")`, not the doubled form.

**Comment by Diyaaa-12** [comment]
Hi @collinsezedike, addressed both review comments:

- fromStroops max BigInt test now asserts the actual output value
- shortenAddress now guards against short inputs and returns the address unchanged when it's too short to truncate

All 45 tests passing!

**Review by collinsezedike — CHANGES_REQUESTED** [review]
Thank you for your time and effort contributing to this project, @Diyaaa-12. The coverage is solid and the tests are cleeeean.

One heads-up: a Prettier formatting pass was recently merged to main as part of the CI quality gates (#308), which unfortunately introduced a conflict. Kindly rebase against main, resolve the conflict, then run `pnpm format` before pushing so the new format check passes.

**Comment by Diyaaa-12** [comment]
Closing in favour of a cleaner PR — see #311
